### PR TITLE
[lldb] Subtract module start address when building addr to module map

### PR DESCRIPTION
**Pull request by augusto2112**
When building the LLDBMemoryReader module to address map, it was assumed that a module would start at address 0, when this was not necessarily true. If many modules had a high start file address, the module map could potentially use up all the available space, and potentially clash with the pointer authentication mask. This patch accounts subtracts the module's start address when building the map, and also verifies that the addresses handed out can't be affected by the pointer authentication mask.

rdar://106055569

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp [hunk 1]
```diff
@@ -502,15 +534,24 @@ LLDBMemoryReader::getFileAddressAndModuleForTaggedAddress(
...
     // Since this is the first registered module,
     // clearing the tag bit will give the virtual file address.
     file_address = tagged_address & ~LLDB_FILE_ADDRESS_BIT;
   else
     // The end of the previous section is the start of the current one.
+    // We also need to add the first section's file address since we remove it
+    // when constructing the range to module map.
     file_address = tagged_address - std::prev(pair_iterator)->first;
 
+  file_address += section_list->GetSectionAtIndex(0)->GetFileAddress();
```
Comment (adrian-prantl) [inline lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp:556]: This wasn't there before, should we comment why we're adding the starting address of [0]?

**Review by adrian-prantl — APPROVED** [review]

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test windows

**Pull request merged**